pkg/flink/internal/store/store.go

```diff
@@ -74,23 +78,42 @@ func (s *Store) ProcessStatement(statement string) (*types.ProcessedStatement, *
 		return result, sErr
 	}
 
+	statementName := uuid.New().String()[:18]
+
 	// Process remote statements
+	computePoolId := s.appOptions.GetComputePoolId()
+	properties := s.Properties.GetSqlProperties()
+
+	var principal string
+	serviceAccount := s.Properties.Get(config.ConfigKeyServiceAccount)
+	if serviceAccount != "" {
+		principal = serviceAccount
+	} else {
+		principal = s.appOptions.GetContext().GetUser().ResourceId
+	}
+
 	statementObj, err := s.authenticatedGatewayClient().CreateStatement(
-		statement,
-		s.appOptions.GetComputePoolId(),
-		s.Properties.GetSqlProperties(),
-		s.Properties.Get(config.ConfigKeyServiceAcount),
-		s.appOptions.GetIdentityPoolId(),
+		createSqlV1beta1Statement(statement, statementName, computePoolId, properties),
+		principal,
 		s.appOptions.GetEnvironmentId(),
 		s.appOptions.GetOrgResourceId(),
 	)
 	if err != nil {
 		status := statementObj.GetStatus()
 		return nil, types.NewStatementErrorFailureMsg(err, status.GetDetail())
 	}
-	processedStatement := types.NewProcessedStatement(statementObj)
-	processedStatement.ServiceAccount = s.Properties.Get(config.ConfigKeyServiceAcount)
-	return processedStatement, nil
+	return types.NewProcessedStatement(statementObj), nil
+}
+
+func createSqlV1beta1Statement(statement string, statementName string, computePoolId string, properties map[string]string) flinkgatewayv1beta1.SqlV1beta1Statement {
+	return flinkgatewayv1beta1.SqlV1beta1Statement{
+		Name: &statementName,
```

Comment:
```suggestion
		Name: flinkgatewayv1beta1.StringPtr(statementName),
```

Reply:
I don't get this one? I've not problem making a method, but what would be the advantage of doing this rather than using the built-in operator?

Reply:
If you want to change this to a string literal it's easier to do so if you use the SDK helper function. Not a huge deal though!